### PR TITLE
Remove hard coded single oracle info in GUI

### PR DESCRIPTION
**Pull request by benthecarman**

### CODE REVIEW

**Comment by nkohen** [comment]
Should probably fix this while you're at it: https://github.com/bitcoin-s/bitcoin-s/blob/adaptor-dlc/app/gui/src/main/scala/org/bitcoins/gui/dlc/dialog/AcceptDLCDialog.scala#L35

File: app/gui/src/main/scala/org/bitcoins/gui/dlc/dialog/AcceptDLCDialog.scala [hunk 1]
```diff
@@ -38,7 +43,7 @@ class AcceptDLCDialog
     val fromOffer = OracleInfo.fromTLV(offer.tlv.contractInfo.oracleInfo)
     val fromAnnouncement = SingleOracleInfo(announcement)
```
Comment (nkohen) [inline app/gui/src/main/scala/org/bitcoins/gui/dlc/dialog/AcceptDLCDialog.scala]: this still seems to be hard-coded to single though no?

**Review by nkohen — COMMENTED** [review]
Does this include the plotCETsWithOriginalCurve -> plotCETs and json parsing fix on the WalletRoutes?

File: app/gui/src/main/scala/org/bitcoins/gui/dlc/dialog/AcceptDLCDialog.scala (hunk 1 above)
Comment (benthecarman) [reply app/gui/src/main/scala/org/bitcoins/gui/dlc/dialog/AcceptDLCDialog.scala]: This is just what we validate against, fixed

**Comment by benthecarman** [comment]
> Does this include the plotCETsWithOriginalCurve -> plotCETs and json parsing fix on the WalletRoutes?

Yes

**Comment by nkohen** [comment]
I'm still getting
```
[error] java.lang.IllegalArgumentException: Unknown OracleAnnouncementTLV type got BigSizeUInt(42770)
[error] 	at org.bitcoins.core.protocol.tlv.TLVParentFactory.fromBytes(TLV.scala:119)
[error] 	at org.bitcoins.core.protocol.tlv.TLVParentFactory.fromBytes$(TLV.scala:113)
[error] 	at org.bitcoins.core.protocol.tlv.OracleAnnouncementTLV$.fromBytes(TLV.scala:785)
[error] 	at org.bitcoins.core.protocol.tlv.OracleAnnouncementTLV$.fromBytes(TLV.scala:785)
[error] 	at org.bitcoins.crypto.Factory.fromHex(Factory.scala:13)
[error] 	at org.bitcoins.crypto.Factory.apply(Factory.scala:56)
[error] 	at org.bitcoins.commons.serializers.Picklers$.$anonfun$dlcStatusR$5(Picklers.scala:406)
[error] 	at scala.collection.StrictOptimizedIterableOps.map(StrictOptimizedIterableOps.scala:99)
[error] 	at scala.collection.StrictOptimizedIterableOps.map$(StrictOptimizedIterableOps.scala:86)
[error] 	at scala.collection.mutable.ArrayBuffer.map(ArrayBuffer.scala:41)
[error] 	at org.bitcoins.commons.serializers.Picklers$.oracles$lzycompute$1(Picklers.scala:405)
[error] 	at org.bitcoins.commons.serializers.Picklers$.oracles$1(Picklers.scala:405)
[error] 	at org.bitcoins.commons.serializers.Picklers$.oracleOutcome$lzycompute$1(Picklers.scala:418)
[error] 	at org.bitcoins.commons.serializers.Picklers$.oracleOutcome$1(Picklers.scala:410)
[error] 	at org.bitcoins.commons.serializers.Picklers$.$anonfun$dlcStatusR$1(Picklers.scala:516)
[error] 	at upickle.core.Types$Reader$$anon$4.mapNonNullsFunction(Types.scala:72)
[error] 	at upickle.core.Visitor$MapReader.$anonfun$visitObject$1(Visitor.scala:198)
[error] 	at upickle.core.Visitor$MapObjContext.visitEnd(Visitor.scala:231)
[error] 	at ujson.AstTransformer.transformObject(AstTransformer.scala:21)
[error] 	at ujson.AstTransformer.transformObject$(AstTransformer.scala:14)
[error] 	at ujson.Value$.transformObject(Value.scala:123)
[error] 	at ujson.Value$.transform(Value.scala:189)
[error] 	at ujson.Value.transform(Value.scala:112)
[error] 	at ujson.Value.transform$(Value.scala:112)
[error] 	at ujson.Obj.transform(Value.scala:228)
[error] 	at upickle.Api.read(Api.scala:32)
[error] 	at upickle.Api.read$(Api.scala:32)
[error] 	at upickle.default$.read(Api.scala:102)
[error] 	at org.bitcoins.gui.dlc.DLCPaneModel.$anonfun$getDLCs$1(DLCPaneModel.scala:62)
[error] 	at scala.collection.StrictOptimizedIterableOps.map(StrictOptimizedIterableOps.scala:99)
[error] 	at scala.collection.StrictOptimizedIterableOps.map$(StrictOptimizedIterableOps.scala:86)
[error] 	at scala.collection.mutable.ArrayBuffer.map(ArrayBuffer.scala:41)
[error] 	at org.bitcoins.gui.dlc.DLCPaneModel.getDLCs(DLCPaneModel.scala:62)
[error] 	at org.bitcoins.gui.dlc.DLCPaneModel.updateDLCs(DLCPaneModel.scala:81)
[error] 	at org.bitcoins.gui.dlc.DLCPaneModel.viewDLC(DLCPaneModel.scala:335)
[error] 	at org.bitcoins.gui.dlc.DLCTableView$$anon$9$$anon$10.$anonfun$new$12(DLCTableView.scala:118)
[error] 	at com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:86)
[error] 	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:238)
[error] 	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
[error] 	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
[error] 	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
[error] 	at com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
[error] 	at com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:49)
[error] 	at javafx.event.Event.fireEvent(Event.java:198)
[error] 	at javafx.scene.control.MenuItem.fire(MenuItem.java:462)
[error] 	at com.sun.javafx.scene.control.skin.ContextMenuContent$MenuItemContainer.doSelect(ContextMenuContent.java:1405)
[error] 	at com.sun.javafx.scene.control.skin.ContextMenuContent$MenuItemContainer.lambda$createChildren$599(ContextMenuContent.java:1358)
[error] 	at com.sun.javafx.event.CompositeEventHandler$NormalEventHandlerRecord.handleBubblingEvent(CompositeEventHandler.java:218)
[error] 	at com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:80)
[error] 	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:238)
[error] 	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
[error] 	at com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(CompositeEventDispatcher.java:59)
[error] 	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
[error] 	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
[error] 	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
[error] 	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
[error] 	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
[error] 	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
[error] 	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
[error] 	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
[error] 	at com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
[error] 	at com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:54)
[error] 	at javafx.event.Event.fireEvent(Event.java:198)
[error] 	at javafx.scene.Scene$MouseHandler.process(Scene.java:3757)
[error] 	at javafx.scene.Scene$MouseHandler.access$1500(Scene.java:3485)
[error] 	at javafx.scene.Scene.impl_processMouseEvent(Scene.java:1762)
[error] 	at javafx.scene.Scene$ScenePeerListener.mouseEvent(Scene.java:2494)
[error] 	at com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:394)
[error] 	at com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:295)
[error] 	at java.security.AccessController.doPrivileged(Native Method)
[error] 	at com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleMouseEvent$345(GlassViewEventHandler.java:432)
[error] 	at com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(QuantumToolkit.java:389)
[error] 	at com.sun.javafx.tk.quantum.GlassViewEventHandler.handleMouseEvent(GlassViewEventHandler.java:431)
[error] 	at com.sun.glass.ui.View.handleMouseEvent(View.java:555)
[error] 	at com.sun.glass.ui.View.notifyMouse(View.java:937)
[error] 	at com.sun.glass.ui.gtk.GtkApplication._runLoop(Native Method)
[error] 	at com.sun.glass.ui.gtk.GtkApplication.lambda$null$203(GtkApplication.java:139)
[error] 	at java.lang.Thread.run(Thread.java:748)
```

**Comment by benthecarman** [comment]
> I'm still getting
> 
> ```
> [error] java.lang.IllegalArgumentException: Unknown OracleAnnouncementTLV type got BigSizeUInt(42770)
> [error] 	at org.bitcoins.core.protocol.tlv.TLVParentFactory.fromBytes(TLV.scala:119)
> [error] 	at org.bitcoins.core.protocol.tlv.TLVParentFactory.fromBytes$(TLV.scala:113)
> [error] 	at org.bitcoins.core.protocol.tlv.OracleAnnouncementTLV$.fromBytes(TLV.scala:785)
> [error] 	at org.bitcoins.core.protocol.tlv.OracleAnnouncementTLV$.fromBytes(TLV.scala:785)
> [error] 	at org.bitcoins.crypto.Factory.fromHex(Factory.scala:13)
> [error] 	at org.bitcoins.crypto.Factory.apply(Factory.scala:56)
> [error] 	at org.bitcoins.commons.serializers.Picklers$.$anonfun$dlcStatusR$5(Picklers.scala:406)
> [error] 	at scala.collection.StrictOptimizedIterableOps.map(StrictOptimizedIterableOps.scala:99)
> [error] 	at scala.collection.StrictOptimizedIterableOps.map$(StrictOptimizedIterableOps.scala:86)
> [error] 	at scala.collection.mutable.ArrayBuffer.map(ArrayBuffer.scala:41)
> [error] 	at org.bitcoins.commons.serializers.Picklers$.oracles$lzycompute$1(Picklers.scala:405)
> [error] 	at org.bitcoins.commons.serializers.Picklers$.oracles$1(Picklers.scala:405)
> [error] 	at org.bitcoins.commons.serializers.Picklers$.oracleOutcome$lzycompute$1(Picklers.scala:418)
> [error] 	at org.bitcoins.commons.serializers.Picklers$.oracleOutcome$1(Picklers.scala:410)
> [error] 	at org.bitcoins.commons.serializers.Picklers$.$anonfun$dlcStatusR$1(Picklers.scala:516)
> [error] 	at upickle.core.Types$Reader$$anon$4.mapNonNullsFunction(Types.scala:72)
> [error] 	at upickle.core.Visitor$MapReader.$anonfun$visitObject$1(Visitor.scala:198)
> [error] 	at upickle.core.Visitor$MapObjContext.visitEnd(Visitor.scala:231)
> [error] 	at ujson.AstTransformer.transformObject(AstTransformer.scala:21)
> [error] 	at ujson.AstTransformer.transformObject$(AstTransformer.scala:14)
> [error] 	at ujson.Value$.transformObject(Value.scala:123)
> [error] 	at ujson.Value$.transform(Value.scala:189)
> [error] 	at ujson.Value.transform(Value.scala:112)
> [error] 	at ujson.Value.transform$(Value.scala:112)
> [error] 	at ujson.Obj.transform(Value.scala:228)
> [error] 	at upickle.Api.read(Api.scala:32)
> [error] 	at upickle.Api.read$(Api.scala:32)
> [error] 	at upickle.default$.read(Api.scala:102)
> [error] 	at org.bitcoins.gui.dlc.DLCPaneModel.$anonfun$getDLCs$1(DLCPaneModel.scala:62)
> [error] 	at scala.collection.StrictOptimizedIterableOps.map(StrictOptimizedIterableOps.scala:99)
> [error] 	at scala.collection.StrictOptimizedIterableOps.map$(StrictOptimizedIterableOps.scala:86)
> [error] 	at scala.collection.mutable.ArrayBuffer.map(ArrayBuffer.scala:41)
> [error] 	at org.bitcoins.gui.dlc.DLCPaneModel.getDLCs(DLCPaneModel.scala:62)
> [error] 	at org.bitcoins.gui.dlc.DLCPaneModel.updateDLCs(DLCPaneModel.scala:81)
> [error] 	at org.bitcoins.gui.dlc.DLCPaneModel.viewDLC(DLCPaneModel.scala:335)
> [error] 	at org.bitcoins.gui.dlc.DLCTableView$$anon$9$$anon$10.$anonfun$new$12(DLCTableView.scala:118)
> [error] 	at com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:86)
> [error] 	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:238)
> [error] 	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
> [error] 	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
> [error] 	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
> [error] 	at com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
> [error] 	at com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:49)
> [error] 	at javafx.event.Event.fireEvent(Event.java:198)
> [error] 	at javafx.scene.control.MenuItem.fire(MenuItem.java:462)
> [error] 	at com.sun.javafx.scene.control.skin.ContextMenuContent$MenuItemContainer.doSelect(ContextMenuContent.java:1405)
> [error] 	at com.sun.javafx.scene.control.skin.ContextMenuContent$MenuItemContainer.lambda$createChildren$599(ContextMenuContent.java:1358)
> [error] 	at com.sun.javafx.event.CompositeEventHandler$NormalEventHandlerRecord.handleBubblingEvent(CompositeEventHandler.java:218)
> [error] 	at com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:80)
> [error] 	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:238)
> [error] 	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
> [error] 	at com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(CompositeEventDispatcher.java:59)
> [error] 	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
> [error] 	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
> [error] 	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
> [error] 	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
> [error] 	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
> [error] 	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
> [error] 	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
> [error] 	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
> [error] 	at com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
> [error] 	at com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:54)
> [error] 	at javafx.event.Event.fireEvent(Event.java:198)
> [error] 	at javafx.scene.Scene$MouseHandler.process(Scene.java:3757)
> [error] 	at javafx.scene.Scene$MouseHandler.access$1500(Scene.java:3485)
> [error] 	at javafx.scene.Scene.impl_processMouseEvent(Scene.java:1762)
> [error] 	at javafx.scene.Scene$ScenePeerListener.mouseEvent(Scene.java:2494)
> [error] 	at com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:394)
> [error] 	at com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:295)
> [error] 	at java.security.AccessController.doPrivileged(Native Method)
> [error] 	at com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleMouseEvent$345(GlassViewEventHandler.java:432)
> [error] 	at com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(QuantumToolkit.java:389)
> [error] 	at com.sun.javafx.tk.quantum.GlassViewEventHandler.handleMouseEvent(GlassViewEventHandler.java:431)
> [error] 	at com.sun.glass.ui.View.handleMouseEvent(View.java:555)
> [error] 	at com.sun.glass.ui.View.notifyMouse(View.java:937)
> [error] 	at com.sun.glass.ui.gtk.GtkApplication._runLoop(Native Method)
> [error] 	at com.sun.glass.ui.gtk.GtkApplication.lambda$null$203(GtkApplication.java:139)
> [error] 	at java.lang.Thread.run(Thread.java:748)
> ```

Fixed, one of the writers was using the `OracleInfo` tlv instead of the announcement

**Review by nkohen — APPROVED** [review]

**Pull request merged**